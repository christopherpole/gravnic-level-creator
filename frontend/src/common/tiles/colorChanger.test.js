import React from 'react';
import { ENTITIES } from 'gravnic-game';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ColorChangerTile from './colorChanger';

configure({ adapter: new Adapter() });

describe('<ColorChangerTile />', () => {
  let props;

  beforeEach(() => {
    props = {
      targetEntity: {
        entityId: ENTITIES.BLOCK.id,
        color: '#ff0000',
      },
    };
  });

  it('Renders without exploding', () => {
    const colorChanger = shallow(<ColorChangerTile {...props} />);

    expect(colorChanger).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const colorChanger = shallow(<ColorChangerTile {...props} />);

    expect(toJson(colorChanger)).toMatchSnapshot();
  });
});
