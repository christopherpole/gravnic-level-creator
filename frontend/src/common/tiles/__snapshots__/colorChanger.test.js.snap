// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorChangerTile /> Matches the current snapshot 1`] = `
<styled.div
  color="#ff0000"
  entityId="BLOCK"
/>
`;
